plugins/jobs/src/js/types/JobConnection.ts

```diff
@@ -1,19 +1,109 @@
-import { Job } from "#PLUGINS/jobs/src/js/types/Job";
+import { Job, JobTypeResolver } from "#PLUGINS/jobs/src/js/types/Job";
+import { JobResponse as MetronomeJobResponse } from "src/js/events/MetronomeClient";
 
+import JobStates from "#PLUGINS/jobs/src/js/constants/JobStates";
+import JobStatus from "#PLUGINS/jobs/src/js/constants/JobStatus";
 export interface JobConnection {
+  path: string[];
   filteredCount: number;
   totalCount: number;
   nodes: Job[];
 }
 
 export const JobConnectionSchema = `
 type JobConnection {
+  path: [String]!
   filteredCount: Int!
   totalCount: Int!
   nodes: [Job]!
 }
 `;
 
-export function JobConnectionTypeResolver(response: any): any {
-  return response;
+export interface JobsQueryArgs {
+  path: string[];
+  filter?: string;
+  sortBy?: SortOption;
+  sortDirection?: SortDirection;
+}
+
+export type SortOption = "ID" | "STATUS" | "LAST_RUN";
+
+export type SortDirection = "ASC" | "DESC";
+
+export function JobConnectionTypeResolver(
+  response: MetronomeJobResponse[],
+  args: JobsQueryArgs
+): JobConnection {
+  const { sortBy = "ID", sortDirection = "ASC", filter = "", path = [] } = args;
+
+  const namespacedResponse: MetronomeJobResponse[] =
+    path.length > 0
```

Comment:
No joke, but I found the ternary rule I would propose adding:
https://eslint.org/docs/rules/multiline-ternary
`never`

Reply:
Discussion for that is on sync. https://jira.mesosphere.com/browse/DCOS-38625

Reply:
Ähem. https://github.com/dcos/dcos-ui/pull/3004#discussion_r194372319